de.metas.acct.base/src/main/sql/postgresql/ddl/functions/AccountSheetReport.sql

```diff
@@ -0,0 +1,235 @@
+DROP FUNCTION IF EXISTS AccountSheetReport(p_dateFrom date, p_dateTo date, p_c_acctschema_id NUMERIC, p_ad_org_id numeric, p_account_id NUMERIC, p_c_activity_id numeric, p_c_project_id numeric);
+
+/*
+- DateFrom/To - mandatory
+- C_AcctSchema_ID - mandatory
+- AD_Org_ID - mandatory
+- Account_ID - optional
+- C_Activity_ID - optional
+- C_Project_ID - optional
+*/
+CREATE OR REPLACE FUNCTION AccountSheetReport(p_dateFrom date,
+                                              p_dateTo date,
+                                              p_c_acctschema_id NUMERIC,
+                                              p_ad_org_id numeric,
+                                              p_account_id NUMERIC=NULL,
+                                              p_c_activity_id numeric=NULL,
+                                              p_c_project_id numeric=NULL)
+    RETURNS table
+            (
+                dateacct         timestamp WITHOUT TIME ZONE,
+                documentno       text,
+                description      text,
+                amtacctdr        numeric,
+                amtacctcr        numeric,
+                account_id       numeric,
+                fact_acct_id     numeric,
+                c_doctype_id     numeric,
+                c_tax_id         numeric,
+                c_taxcategory_id numeric,
+                beginningBalance numeric,
+                endingBalance    numeric
+            )
+AS
+$BODY$
+DECLARE
+    v_time                              timestamp;
+    LINE_TYPE_BEGINNINGBALANCE CONSTANT text = 'B';
+    LINE_TYPE_TRANSACTION      CONSTANT text = 'T';
+    v_temp                              numeric;
+BEGIN
+    v_time := logDebug('start');
+
+    --
+    -- create temporary table for everything we're working on; it has no rows, only the needed columns
+    DROP TABLE IF EXISTS tbpFilteredFactAcct;
+    CREATE TEMPORARY TABLE tbpFilteredFactAcct AS
```

Comment:
use `TMP_AccountSheetReport` tablename instead of `tbpFilteredFactAcct`

Reply:
renamed